ui/helpers/utils/token-util.js

```diff
@@ -226,7 +226,12 @@ export function getTokenFiatAmount(
     .times(new Numeric(conversionRate, 10))
     .toString();
 
-  let currentTokenInFiat = new Numeric(tokenAmount, 10);
+  // the `string` property of a token is typically passed as `tokenAmount`; for very low values, it will be "< 0.000001"
+  const isTokenAmountNearZero = tokenAmount.includes('<');
```

Comment:
`tokenAmount` can be a string or a number; this throws an error if it's the latter. Moreover, adding this edge case here seems like a fragile way to handle small token amounts

I'd recommend removing `string: primaryCurrencyProperties.value` in asset-picker-modal, to avoid this edge case here and in useTokensToSearch. Doing so causes the fallback string representation to show up, which seems acceptable
![Screenshot 2024-05-14 at 3 54 36 PM](https://github.com/MetaMask/metamask-extension/assets/100321200/c824aadb-0839-4b40-ad18-8f1ff84234b8)

Reply:
I think this had to do with how I was reproing this issue. Mind checking if you see it on the latest branch?

Reply:
LGTM